lib/datadog/core/configuration/option.rb

```diff
@@ -102,6 +191,33 @@ def context_eval(&block)
           @context.instance_eval(&block)
         end
 
+        def set_value_from_env_or_default
+          value = nil
+          precedence = nil
+
+          if definition.env && ENV[definition.env]
+            value = coerce_env_variable(ENV[definition.env])
+            precedence = Precedence::PROGRAMMATIC
+          end
+
+          if value.nil? && definition.deprecated_env && ENV[definition.deprecated_env]
+            value = coerce_env_variable(ENV[definition.deprecated_env])
+            precedence = Precedence::PROGRAMMATIC
+
+            Datadog::Core.log_deprecation do
+              "#{definition.deprecated_env} environment variable is deprecated, use #{definition.env} instead."
+            end
+          end
+
+          option_value = value.nil? ? default_value : value
+
+          set(option_value, precedence: precedence || Precedence::DEFAULT)
+        end
+
+        def skip_validation?
```

Comment:
Maybe not for this PR, but it may be worth using an `OnlyOnce` to drop a warning to customers when we detect this is being used. This way we push people to stop using it, as well as are able to see it if someone reports an issue and shares a log.